.jenkinsci/utils/docker-pull-or-build.groovy

```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env groovy
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+//
+// This module helps automatically build new docker develop-build image if Dockerfile changed
+//
+
+def buildOptionsString(options) {
+  def s = ''
+  if (options) {
+    options.each { k, v ->
+      s += "--build-arg ${k}=${v} "
+    }
+  }
+  return s
+}
+
+def dockerPullOrBuild(imageName, currentDockerfileURL, referenceDockerfileURL, scmVars, environment, forceBuild=false, buildOptions=null) {
+  buildOptions = buildOptionsString(buildOptions)
+  withEnv(environment) {
+    def utils = load '.jenkinsci/utils/utils.groovy'
+    sh("docker pull ${env.DOCKER_REGISTRY_BASENAME}:${imageName} || true")
+    randDir = sh(script: "cat /dev/urandom | tr -dc 'a-zA-Z0-9' | head -c 10", returnStdout: true).trim()
+    currentDockerfile = utils.getUrl(currentDockerfileURL, "/tmp/${randDir}/currentDockerfile", true)
+    referenceDockerfile = utils.getUrl(referenceDockerfileURL, "/tmp/${randDir}/referenceDockerfile")
```

Comment:
If we push new Dockerfile to master, will it build docker image?
because `currentDockerfile` and `referenceDockerfile` will be same.

Reply:
Yes, because we always build Docker image regardless of the changes in Dockerfiles:
https://github.com/hyperledger/iroha/blob/3235c4764c84955aa3b5dfbb0712287e0d9928ea/.jenkinsci/utils/docker-pull-or-build.groovy#L36

Reply:
It will significantly increase build time?

Reply:
It will not since we pull an image first. And in most cases we get 100% cache hit